src/CommandEvent.cc

```diff
@@ -77,106 +76,152 @@ CommandEvent::CommandEvent() : Event() {
 }
 
 
+/** private method to initialize the CommandEvent
+  */
 void CommandEvent::_init() {
-    _keywords.insert(ORIGINATORID);
-    _keywords.insert(ORIG_LOCALID);
+    _keywords.insert(ORIG_HOSTNAME);
     _keywords.insert(ORIG_PROCESSID);
-    _keywords.insert(ORIG_IPID);
+    _keywords.insert(ORIG_LOCALID);
 
-    _keywords.insert(DESTINATIONID);
-    _keywords.insert(DEST_LOCALID);
+    _keywords.insert(DEST_HOSTNAME);
     _keywords.insert(DEST_PROCESSID);
-    _keywords.insert(DEST_IPID);
+    _keywords.insert(DEST_LOCALID);
 }
 
+/**
+  * \brief Constructor for CommandEvent
+  * \param msg a cms::TextMessage to convert into a CommandEvent
+  */
 CommandEvent::CommandEvent(cms::TextMessage *msg) : Event(msg) {
     _init();
 
 
-    _psp->set(ORIGINATORID, (int64_t)msg->getLongProperty(ORIGINATORID));
-    _psp->set(ORIG_LOCALID, (short)msg->getShortProperty(ORIG_LOCALID));
+    _psp->set(ORIG_HOSTNAME, (std::string)msg->getStringProperty(ORIG_HOSTNAME));
     _psp->set(ORIG_PROCESSID, (int)msg->getIntProperty(ORIG_PROCESSID));
-    _psp->set(ORIG_IPID, (int)msg->getIntProperty(ORIG_IPID));
+    _psp->set(ORIG_LOCALID, (int)msg->getIntProperty(ORIG_LOCALID));
 
-    _psp->set(DESTINATIONID, (int64_t)msg->getLongProperty(DESTINATIONID));
-    _psp->set(DEST_LOCALID, (short)msg->getShortProperty(DEST_LOCALID));
+    _psp->set(DEST_HOSTNAME, (std::string)msg->getStringProperty(DEST_HOSTNAME));
     _psp->set(DEST_PROCESSID, (int)msg->getIntProperty(DEST_PROCESSID));
-    _psp->set(DEST_IPID, (int)msg->getIntProperty(DEST_IPID));
+    _psp->set(DEST_LOCALID, (int)msg->getIntProperty(DEST_LOCALID));
 
 }
+/**
+  * \brief Constructor for CommandEvent
+  * \param originator originating location of this event
+  * \param destination destination location for this event
+  * \param psp PropertySet to pass in this event
+  */
+CommandEvent::CommandEvent(const LocationID&  originator, const LocationID& destination, const PropertySet::Ptr psp) : Event(*psp) {
```

Comment:
This documentation should be in the header file.
Need a better name than `psp` (don't name after the type).